SNIP-52.md

```diff
@@ -0,0 +1,653 @@
+
+
+### How it works (high-level)
+
+First, let's review the basic components that make this possible:
+
+1. The existing [Tendermint event stack](https://docs.tendermint.com/v0.34/tendermint-core/subscription.html) is a [publish-subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern) model that allows nodes to transmit network events directly to subscribed clients. It does so using JSONRPC over WebSockets.
+
+2. Leveraging the [`add_attribute_plaintext(...)`](https://github.com/scrtlabs/SecretNetwork/blob/ddb9b67a23a2eb5f64076fa3e0a836d186a02b63/cosmwasm/contracts/v1/compute-tests/test-compute-contract/src/contract.rs#L60C1-L62) API method, Secret Contracts can add custom key-value attributes to the transaction's log.
+
+3. Clients coordinate with smart contracts to determine globally unique, single-use "Notification IDs" which represent specific future events. When one of those specific events occur, its Notication ID is added to the transaction log as a custom attribute. Since the contract and recipient are the only parties aware of a Notification ID's significance, the event in that transaction's log is what discreetly notifies the recipient, effectively creating a private push-based notification service.
+
+
+### Example flow
+
+Let's walk through a simple example, where client Alice wants to be notified next time she receives a transfer of token X into her account.
+
+> NOTE: Example uses fake base64 data for brevity
+
+1. Alice queries the token X contract to get the unique Notification ID for her next incoming transfer:
+    ```json
+    {
+      "channel_info": {
+        "channel": "transfers"
+      }
+    }
+    ```
+
+2. The contract responds:
+    ```json
+    {
+      "channel_info": {
+        "channel": "transfers",
+        "seed": "ecc7f60418aa",
+        "counter": "3",
+        "next_id": "ZjZjYzVhYjU4",
+        "as_of_block": "1131420"
+      }
+    }
+    ```
+
+    Alice now has a globally unique Notification ID the contract will use next time someone transfers tokens to her account.
+
+    Furthermore, Alice now has a seed she can use to derive future Notification IDs offline for subsequent transfer events (i.e., without having to query this contract again)
+
+3. Alice subscribes to execution events on the token contract:
+    ```json
+    {
+      "jsonrpc": "2.0",
+      "id": "0",
+      "method": "subscribe",
+      "params": {
+        "query": "wasm.contract_address='secret1ku936rhzqynw6w2gqp4e5hsdtnvwajj6r7mc5z'"
+      }
+    }
+    ```
+
+    Alice will now receive a JSONRPC message for each new execution of the contract, from which she can search for her unique Notification ID.
+    
+    <!-- Alternatively, if she trusts the WebSocket server won't record her activity, she can use a filter in the `query` field of her subscription message, e.g., `wasm.ZjZjYzVhYjU4 EXISTS` . -->
```

Comment:
Why is this commented out? It does not appear in the markdown rendering of the document

Reply:
Thanks, it was put on hold during the transition to Counter vs TxHash mode. It has been updated.